### PR TITLE
Dart 2 fixes

### PR DESCRIPTION
**Pull request by jakobr-google**
Fixes runtime type errors in Dart 2 mode.

### CODE REVIEW

File: lib/src/auth/auth.dart [hunk 1]
```diff
@@ -29,7 +29,7 @@ abstract class BaseAuthenticator {
   auth.AccessToken _accessToken;
   String _lastUri;
 
-  Future authenticate(Map<String, String> metadata, String uri) async {
+  Future<Null> authenticate(Map<String, String> metadata, String uri) async {
```
Comment (mravn-google) [inline lib/src/auth/auth.dart]: Can't you use `Future<void>` here yet?

File: lib/src/server/service.dart [hunk 2]
```diff
@@ -34,6 +37,56 @@ class ServiceMethod<Q, R> {
       this.streamingResponse,
       this.requestDeserializer,
       this.responseSerializer);
+
+  StreamController<Q> createRequestStream(StreamSubscription incoming) =>
+      new StreamController(
```
Comment (mravn-google) [inline lib/src/server/service.dart]: Is `new StreamController<Q>` inferred here?

File: lib/src/auth/auth.dart (hunk 1 above)
Comment (jakobr-google) [reply lib/src/auth/auth.dart]: I'd like to keep this code running on latest stable Dart SDK as long as possible, so not yet...

File: lib/src/server/service.dart (hunk 2 above)
Comment (jakobr-google) [reply lib/src/server/service.dart]: I think so - it works, or at least I don't get the runtime exception anymore. :)

I'll make it explicit.

**Pull request merged**